packages/interactivity/src/directives.js

```diff
@@ -278,4 +285,55 @@ export default () => {
 			} );
 		}
 	);
+
+	// data-wp-slot
+	directive(
+		'slot',
+		( {
+			directives: {
+				slot: { default: slot, above, below },
+			},
+			props: { children },
+		} ) => {
+			return (
+				<Fragment>
+					{ above && <Slot name={ above } /> }
+					{ slot ? (
+						<Slot name={ slot }>{ children }</Slot>
+					) : (
+						children
+					) }
+					{ below && <Slot name={ below } /> }
+				</Fragment>
+			);
+		},
+		{ priority: 4 }
+	);
+
+	// data-wp-fill
+	directive(
+		'fill',
+		( {
+			directives: {
+				fill: { default: fill },
+			},
+			props: { children },
+			evaluate,
+			context,
+		} ) => {
+			const contextValue = useContext( context );
+			const slot = evaluate( fill, { context: contextValue } );
+			return <Fill slot={ slot }>{ children }</Fill>;
+		},
+		{ priority: 4 }
```

Comment:
@darerodz: And shouldn't this priority be more than `data-wp-context` so they can be used in the same element?

```html
<div data-wp-context='{ "fill": "some fill name" }' data-wp-fill="context.fill">
  I'm a fill and I can be repositioned by changing `context.fill`
</div>
```

Reply:
As this directive adds a wrapper around the main element, I felt it would be better to give more priority to these directives than regular directives. But you're right; `data-wp-context` should be evaluated first if you want to use it in the slot container.

I'll consider it when I move the slot & fill implementation to its own PR.

Reply:
As I mentioned in https://github.com/WordPress/gutenberg/pull/53958#issuecomment-1696049861:

> [...] this is something I wanted to change. Surprisingly, the implementation stopped working as expected if I changed the priority order, so I decided to leave the current values for now.

For some reason, the `Slot` component behaves weirdly if I change the directive priorities. When it adds the slot before or after the main element, the `Slot` fallback is duplicated, and the `Fill` disappears.

I tried to figure out why that was happening without success, so I eventually decided to leave the current priority values, which are working fine. We can take a deeper look at this in the future.

cc: @luisherranz